src/evaluation/mod.rs

```diff
@@ -0,0 +1,79 @@
+use std::collections::HashMap;
+
+#[derive(Clone)]
+pub struct EvaluationContext {
+    pub targetting_key: String,
+    pub attributes: HashMap<String, String>,
+}
+
+pub struct EvaluationOptions {}
+
+pub type FlattenedContext = HashMap<String, String>;
+
+pub fn flatten_context(context: EvaluationContext) -> FlattenedContext {
+    let mut flattened_context = HashMap::new();
+    flattened_context.insert(
+        context.targetting_key(),
+        context.attribute(context.targetting_key()),
+    );
+    for (key, value) in context.attributes() {
+        flattened_context.insert(key, value);
+    }
+    flattened_context
+}
+// impl
+impl EvaluationContext {
+    pub fn new(targetting_key: String, attributes: HashMap<String, String>) -> Self {
+        Self {
+            targetting_key,
+            attributes,
+        }
+    }
+    fn attribute(&self, key: String) -> String {
```

Comment:
Is this idiomatic for the language? I'd expect something like `get`?

Reply:
Done